appcues/src/main/java/com/appcues/ui/AppcuesOverlayViewManager.kt

```diff
@@ -1,195 +1,129 @@
 package com.appcues.ui
 
-import android.annotation.SuppressLint
 import android.app.Activity
-import android.os.Build.VERSION
-import android.os.Build.VERSION_CODES
 import android.view.View
 import android.view.ViewGroup
-import android.view.inspector.WindowInspector
+import android.widget.FrameLayout.LayoutParams
 import androidx.compose.runtime.remember
 import androidx.compose.ui.platform.ComposeView
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
 import androidx.core.view.children
 import androidx.lifecycle.DefaultLifecycleObserver
 import androidx.lifecycle.LifecycleOwner
-import androidx.lifecycle.ViewTreeLifecycleOwner
-import androidx.lifecycle.ViewTreeViewModelStoreOwner
 import androidx.lifecycle.findViewTreeLifecycleOwner
-import androidx.lifecycle.findViewTreeViewModelStoreOwner
-import androidx.savedstate.findViewTreeSavedStateRegistryOwner
-import androidx.savedstate.setViewTreeSavedStateRegistryOwner
 import com.appcues.R
 import com.appcues.data.model.RenderContext
-import com.appcues.databinding.AppcuesOverlayLayoutBinding
-import com.appcues.logging.Logcues
 import com.appcues.monitor.AppcuesActivityMonitor
 import com.appcues.monitor.AppcuesActivityMonitor.ActivityMonitorListener
 import com.appcues.ui.composables.AppcuesComposition
 import com.appcues.ui.primitive.EmbedChromeClient
+import com.appcues.ui.utils.getParentView
 import org.koin.core.scope.Scope
-import java.lang.reflect.Method
 
 internal class AppcuesOverlayViewManager(
     private val scope: Scope,
     private val renderContext: RenderContext
 ) : DefaultLifecycleObserver, ActivityMonitorListener {
 
-    private val logcues: Logcues by lazy { scope.get() }
-
-    private var viewModel: AppcuesViewModel? = null
-
-    private var shakeGestureListener: ShakeGestureListener? = null
+    private lateinit var viewModel: AppcuesViewModel
+    private lateinit var gestureListener: ShakeGestureListener
```

Comment:
setting those as lateinit var will work the same as we had before, but this way we can avoid calling "!!" when creating the Composition

Reply:
What if I background and foreground without ever having an experience shown (no call to `start()`) - wouldn't that end up in `onResume` and reference `gestureListener` before it was initialized, causing an exception?

Reply:
no because even though we are observing the lifecycle we only do it after we create the instances.

https://github.com/appcues/appcues-android-sdk/pull/438/files#diff-574c69d3146da26d9f6652cfdfcc11b4357f3dec4e153f40de758d6306794472R68

Reply:
ah! right, thank you!